posts/2021-10-05-configurable-response-headers-21.0.0.11-beta.adoc

```diff
@@ -0,0 +1,174 @@
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: Configurable Response Headers and more exciting new features in Open Liberty 21.0.0.11-beta - OpenLiberty.io
+seo-description: Open Liberty 21.0.0.11-beta provides new configurable response headers, offering more granular control over response headers, as well as important changes to the Jakarta EE convenience features, with the removal of Jakarta EE 9.0 features in the transition to Jakarta EE 9.1.
+blog_description: Open Liberty 21.0.0.11-beta provides new configurable response headers, offering more granular control over response headers, as well as important changes to the Jakarta EE convenience features, with the removal of Jakarta EE 9.0 features in the transition to Jakarta EE 9.1.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Configurable Response Headers and more exciting new features in Open Liberty 21.0.0.11-beta
+Ryan Storey <https://github.com/ryan-storey>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 21.0.0.11-beta provides new configurable response headers, offering more granular control over response headers, as well as important changes to the Jakarta EE convenience features, with the removal of Jakarta EE 9.0 features in the transition to Jakarta EE 9.1.
+
+We have two beta packages for link:{url-about}[Open Liberty]:
+
+* <<allbeta, All Beta Features>>: a larger package that contains all Open Liberty beta features (including Jakarta EE 9.1 beta features) and GA features and functions.
+* <<jakarta, Jakarta EE 9.1 Beta Features>>: a lightweight package that contains only the Jakarta EE 9.1 features.
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools.
+
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+The All Beta Features package includes the following beta features:
+
+* <<response_headers, Configurable Response Headers>>
+
+[#response_headers]
+=== Configurable Response Headers
+
+You can now configure Open Liberty (21.0.0.11-beta) to modify response headers. The available configuration options allow for headers to be appended, for existing headers to be overwritten, for missing headers to be added, and for undesired headers to be removed from all responses being serviced by an HTTP endpoint. This configuration offers more granular control over response headers, which offers a solution to modifying headers without the need to modify existing applications, filters, or otherwise. 
+
+==== Try it out! 
+
+To use configurable response headers, begin by defining a new element called `<headers>` in the `server.xml`.  You can configure this for individual HTTP endpoints or for all endpoints at once. 
+
+Configuring for individual HTTP endpoints:
```

Comment:
would it be helpful to clarify when you introduce each example that to modify individual endpoints, you can specify the `headers` element within the `httpEndpoint` element?

Reply:
Thanks for the comments David! However, I don't understand what you mean here unfortunately.

Reply:
@dmuelle do you mean something like:

> To use configurable response headers, begin by defining a new element called `headers` in the `server.xml`. You can configure this for individual HTTP endpoints by specifying the `headers` element as a child of `httpEndpoint` or for all endpoints at once by specifying `headers` element as a peer of `httpEndpoint` and then referencing it in each `httpEndpoint` element via `headersRef` attribute.

Reply:
sorry- wasn't particularly clear :) 

There are two options for configuring `headers`, each with an example
1. specify for a particular endpoint- `headers` element is specified as a sub element of `httpEndpoint`
2. specify for all endpoints-  `headers` element is specified as a top-level html element, outside of `httpEndpoint`

I was suggesting to point out the difference when you introduce each example, eg.
 "To configure the `headers` element for an individual endpoint, specify the element inside the `httpEndpoint` configuration for that endpoint:
 _example 1_
 
 To configure the `headers` element to apply to all enpointsnin your configuration, specify it independently as a top-level element:
 _example 2_
 
 
 see what you think- if it makes the config more evident, great. If it just adds more words tom what already seems evident, leave as is. Thanks

Reply:
just saw Michal's comment and updated my suggestion

Reply:
```suggestion
To configure the `headers` element for an individual endpoint, specify the element inside the `httpEndpoint` configuration for that endpoint:
```